ecl/resource_ecl_provider_connectivity_tenant_connection_request_v2.go

```diff
@@ -0,0 +1,274 @@
+	}
+
+	var hasChange bool
+	var updateOpts tenant_connection_requests.UpdateOpts
+
+	if d.HasChange("name") {
+		hasChange = true
+		updateOpts.Name = d.Get("name").(string)
+	}
+
+	if d.HasChange("description") {
+		hasChange = true
+		updateOpts.Description = d.Get("description").(string)
+	}
+
+	if d.HasChange("tags") {
+		hasChange = true
+		updateOpts.Tags = getTags(d, "tags")
+	}
+
+	if d.HasChange("name_other") {
+		hasChange = true
+		updateOpts.NameOther = d.Get("name_other").(string)
+	}
+
+	if d.HasChange("description_other") {
+		hasChange = true
+		updateOpts.DescriptionOther = d.Get("description_other").(string)
+	}
+
+	if d.HasChange("tags_other") {
+		hasChange = true
+		updateOpts.TagsOther = getTags(d, "tags_other")
+	}
+
+	if hasChange {
+		r := tenant_connection_requests.Update(client, d.Id(), updateOpts)
+		if r.Err != nil {
+			return fmt.Errorf("error updating ECL Provider Connectivity Tenant Connection Request: %s", r.Err)
+		}
+		log.Printf("[DEBUG] Tenant Connection Request has successfully updated.")
+	}
+
+	return resourceProviderConnectivityTenantConnectionRequestV2Read(d, meta)
+}
+
+func resourceProviderConnectivityTenantConnectionRequestV2Delete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	if err := tenant_connection_requests.Delete(client, d.Id()).ExtractErr(); err != nil {
+		return fmt.Errorf("error deleting ECL Provider Connectivity Tenant Connection Request: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Target:     []string{"DELETED"},
+		Refresh:    waitForTenantConnectionRequestStateDelete(client, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf("Error deleting ECL tenant connection request (%s): %s", d.Id(), err)
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func getTags(d *schema.ResourceData, tagName string) map[string]string {
+	rawTags := d.Get(tagName).(map[string]interface{})
+	tags := map[string]string{}
+	for key, value := range rawTags {
+		if v, ok := value.(string); ok {
+			tags[key] = v
+		}
+	}
+	return tags
+}
+
+func waitForTenantConnectionRequestCreate(connClient *eclcloud.ServiceClient, sssClient *eclcloud.ServiceClient, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		request, err := tenant_connection_requests.Get(connClient, id).Extract()
+
+		if err == nil && approvalRequestExists(sssClient, request.ApprovalRequestID) {
+			return request, request.Status, nil
+		}
+
+		return nil, "", err
+	}
+}
+
+func approvalRequestExists(sssClient *eclcloud.ServiceClient, approvalRequestID string) bool {
+	_, err := approval_requests.Get(sssClient, approvalRequestID).Extract()
```

Comment:
You can make it shorter.
```suggestion
	if _, err := approval_requests.Get(sssClient, approvalRequestID).Extract(); err != nil {
```

Reply:
I'll do it.

Reply:
Fixed.